resources/user.rb

```diff
@@ -220,52 +220,51 @@ def revokify_key(key)
   test_table = new_resource.database_name ? 'mysql.db' : 'mysql.user'
 
   # Test
-  incorrect_privs = nil
+  privs_to_grant = []
   test_sql = "SELECT * from #{test_table}"
   test_sql << " WHERE User='#{new_resource.username}'"
   test_sql << " AND Host='#{new_resource.host}'"
   test_sql << " AND Db='#{new_resource.database_name}'" if new_resource.database_name
   test_sql_results = run_query test_sql
 
-  incorrect_privs = true if test_sql_results.split("\n").count == 0
-  # These should all be 'Y'
-  unless test_sql_results.split("\n").count <= 1
+  if test_sql_results.split("\n").count == 0
+    privs_to_grant = desired_privs.map { |p| get_clean_grant_right_name(p) }
+  else
+    # These should all be 'Y'
     parsed_result = parse_mysql_batch_result(test_sql_results)
     Chef::Log.debug(parsed_result)
     parsed_result.each do |r|
       desired_privs.each do |p|
         key = p.to_s.capitalize.tr(' ', '_').gsub('Replication_', 'Repl_').gsub('Create_temporary_tables', 'Create_tmp_table').gsub('Show_databases', 'Show_db')
         key = "#{key}_priv"
-        incorrect_privs = true if r[key] != 'Y'
+        privs_to_grant << get_clean_grant_right_name(p) if r[key] != 'Y'
```

Comment:
Unfortunately, it's not working with grant. You cannot add right one by one, like you can do with revoke (you can revoke only 1 right on 4). You must set all rights you want within one GRANT. Not that easy to understand... But it's like that.

So if one right is missing, you have to add all the rights you want in the GRANT request. That's why we used here a boolean.

Reply:
You mean that 2 subsequent queries, for ex. :
"GRANT SELECT"
followed by
"GRANT UPDATE"
won't work as intended ? 
What will be the final state ? Only the last one will be enforced ?

Reply:
I not sure to properly get your point, this code only executes one GRANT request.